resources/assets/js/components/DiscourseTopic.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+    <b-tr class="topic">
+        <b-td class="main-link"><a :href="url">{{ topic.title }}</a>
+            <div>
+                <a class="badge-wrapper" :href="categoryUrl">
+                    <span class="badge-category-bg"
+                          v-bind:style="{ backgroundColor: '#'+topic.category.color}"
+></span>
+                    <span data-drop-close="true" class="badge-category clear-badge" :title="topic.category.title">
+                        <span class="category-name">
+                            {{ topic.category.name }}
+                        </span>
+                    </span>
+                </a>
+            </div>
+        </b-td>
+        <b-td></b-td>
+        <b-td class="text-center">{{ topic.posts_count }}</b-td>
+        <b-td class="text-center">{{ ago }}</b-td>
+    </b-tr>
+</template>
+
+<script>
+import moment from 'moment'
+
+export default {
+  props: {
+    topic: {
+      type: Object,
+      required: true
+    },
+    discourseBaseUrl: {
+      type: String,
+      required: true
+    }
+  },
+  computed: {
+    ago() {
+      return new moment(this.topic.created_at).fromNow()
+    },
+    url() {
+      return this.discourseBaseUrl + '/t/' + this.topic.slug
+    },
+    categoryUrl() {
+      // todo - should be category url, but needs building for child categories
+      //return this.discourseBaseUrl + this.topic.category.topic_url
+      return this.discourseBaseUrl + this.topic.category.topic_url
+    }
+  }
+}
+</script>
+
+<style scoped lang="scss">
+.topic {
+    border: 0;
+    background-color: white;
+    padding: 10px;
+}
+
+.table tr {
+    border: 0 !important;
+
+    &:hover {
+        border: 1px solid #222;
+        box-shadow: 5px 5px 0 0, 1px 1px 0 0 inset, -1px -1px 0 0 inset;
+    }
+}
+
+.table td {
+    border: 0 !important;
+}
+
+td.main-link {
+    word-break: break-word;
+    color: #222;
+
+    a {
+        text-decoration: none;
+
+        &:hover {
+            text-decoration: underline;
+        }
+    }
+}
+
+
+
+.badge-wrapper {
+    font-size: .8706em;
```

Comment:
There's lots of this kind of hard-coded font size, px numbers, stuff in the codebase, and I've added some too.  Horrid, isn't it?

Reply:
I've taken this from the Discourse CSS again, although it is such a remarkably specific number I guess they're calculating it somewhere.  What would be better here?

Reply:
Probably the ideal is to introduce a set of variables, and then use those consistently throughout the code.  But I think it's a bit hopeless at the moment, it really needs a global overhaul as per the colours.